### PR TITLE
Add Ohio non-public school credits

### PR DESCRIPTION
**Pull request by Linfeng-ll**
Add oh_non_public_school_credits variable and test files

### CODE REVIEW

File: changelog_entry.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
+  changes:
+    added:
+    - Added Ohio Non public school credit
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:4]:
```suggestion
    - Ohio Non public school credit.
```

File: policyengine_us/parameters/gov/states/oh/tax/income/credits/non_public_tuition/agi_credit_amount.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+description: OH Nonchartered, Nonpublic, School Tuition Credit AGI Credit Amount
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/oh/tax/income/credits/non_public_tuition/agi_credit_amount.yaml:1]:
```suggestion
description: Ohio provides this Nonchartered, Nonpublic, School Tuition Credit AGI Credit amount.
```

File: policyengine_us/parameters/gov/states/oh/tax/income/credits/non_public_tuition/agi_credit_amount.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+      2021-01-01: 1_000
+  - threshold:
+      2021-01-01: 100_000
+    amount:
+      2021-01-01: 0
+metadata:
+  threshold_unit: currency-USD
+  amount_unit: currency-USD
+  type: single_amount
+  label: OH Non Public School Credit
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/oh/tax/income/credits/non_public_tuition/agi_credit_amount.yaml:19]:
```suggestion
  label: Ohio Non Public School Credit amount
```

File: policyengine_us/variables/gov/states/oh/tax/income/credits/oh_non_public_school_credits.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
+from policyengine_us.model_api import *
+
+
+class oh_non_public_school_credits(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = (
+        "Ohio Nonchartered, Nonpublic, School Tuition Credit AGI Credit Rates"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/oh/tax/income/credits/oh_non_public_school_credits.py:8]:
```suggestion
        "Ohio Nonchartered, Nonpublic, School Tuition Credit AGI Credit"
```

File: changelog_entry.yaml
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: need to run `git pull origin master`

File: policyengine_us/parameters/gov/states/oh/tax/income/credits/non_public_tuition/agi_credit_amount.yaml [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  threshold_unit: currency-USD
+  amount_unit: currency-USD
+  type: single_amount
+  label: Ohio Non Public School Credit amount
+  reference:
+    - title: Ohio Income Tax Form, Line 16 – Nonchartered, Nonpublic School Tuition Credit, R.C. 5747.75
+      href: https://tax.ohio.gov/static/forms/ohio_individual/individual/2021/pit-it1040-booklet.pdf#page=21
+    - title: Credit for dependents who attend nonchartered nonpublic school.
+      href: https://codes.ohio.gov/ohio-revised-code/section-5747.75
+
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/oh/tax/income/credits/non_public_tuition/agi_credit_amount.yaml:25]: add 2022 tax form as well (should be 3 references total)

File: policyengine_us/variables/household/demographic/person/non_public_school_tuition.py
Comment (MaxGhenis) [inline policyengine_us/variables/household/demographic/person/non_public_school_tuition.py:1]: Move to `household/expense/education/`

File: policyengine_us/tests/policy/baseline/gov/states/oh/tax/income/credits/oh_non_public_school_credits.yaml [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+- name: No non public tuition
+  period: 2021
+  input:
+    oh_agi: 80_000
+    non_public_school_tuition: 0
+    state_code: OH
+  output:
+    oh_non_public_school_credits: 0
+
+- name: AGI less that 50_000
```
Comment (MaxGhenis) [inline policyengine_us/tests/policy/baseline/gov/states/oh/tax/income/credits/oh_non_public_school_credits.yaml:19]: Only need underscores in values
```suggestion
- name: AGI less than $50,000
```

File: policyengine_us/variables/gov/states/oh/tax/income/credits/oh_non_public_school_credits.py [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+    definition_period = YEAR
+    reference = "https://tax.ohio.gov/static/forms/ohio_individual/individual/2021/pit-it1040-booklet.pdf#page=21"
+    defined_for = StateCode.OH
+
+    def formula(tax_unit, period, parameters):
+        rates = parameters(
+            period
+        ).gov.states.oh.tax.income.credits.non_public_tuition.agi_credit_amount
+
+        agi = tax_unit("oh_agi", period)
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/oh/tax/income/credits/oh_non_public_school_credits.py:18]: The law says it's based on federal AGI
```suggestion
        agi = tax_unit("adjusted_gross_income", period)
```

File: policyengine_us/parameters/gov/states/oh/tax/income/credits/non_public_tuition/agi_credit_amount.yaml [hunk 8]
```diff
@@ -0,0 +1,27 @@
+description: Ohio provides this Nonchartered, Nonpublic, School Tuition Credit AGI Credit amount.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/oh/tax/income/credits/non_public_tuition/agi_credit_amount.yaml:1]:
```suggestion
description: Ohio caps the Nonchartered, Nonpublic, School Tuition Credit at this amount, depending on federal adjusted gross income.
```

File: policyengine_us/variables/gov/states/oh/tax/income/credits/oh_non_public_school_credits.py [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+        person = tax_unit.members
+        tuitions = sum(person("non_public_school_tuition", period))
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/oh/tax/income/credits/oh_non_public_school_credits.py:20]: Please add unit tests to ensure it's being capped by tuition expenses (as it currently isn't)
```suggestion
        tuition = tax_unit.sum(person("non_public_school_tuition", period))
        cap = rates.calc(agi)
        return min_(tuition, cap)
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_us/tests/policy/baseline/gov/states/oh/tax/income/credits/oh_non_public_school_credits.yaml [hunk 10]
```diff
@@ -1,44 +1,44 @@
...
   input:
-    oh_agi: 49_999
-    non_public_school_tuition: 1000
+    adjusted_gross_income: 49_999
+    non_public_school_tuition: 1_000
     state_code: OH
   output:
     oh_non_public_school_credits: 500
 
 - name: AGI greater that 50_000 case 1
```
Comment (MaxGhenis) [inline policyengine_us/tests/policy/baseline/gov/states/oh/tax/income/credits/oh_non_public_school_credits.yaml:28]: also reformat per above (and the below one too)

File: policyengine_us/variables/gov/states/oh/tax/income/credits/oh_non_public_school_credits.py [hunk 11]
```diff
@@ -13,10 +13,10 @@ class oh_non_public_school_credits(Variable):
     def formula(tax_unit, period, parameters):
         rates = parameters(
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/oh/tax/income/credits/oh_non_public_school_credits.py:14]: This isn't a rate.
```suggestion
        p = parameters(
```

File: policyengine_us/variables/gov/states/oh/tax/income/credits/oh_non_public_school_credits.py [hunk 12]
```diff
@@ -13,10 +13,10 @@ class oh_non_public_school_credits(Variable):
...
             period
-        ).gov.states.oh.tax.income.credits.non_public_tuition.agi_credit_amount
+        ).gov.states.oh.tax.income.credits.non_public_tuition
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/oh/tax/income/credits/oh_non_public_school_credits.py:16]:
```suggestion
        ).gov.states.oh.tax.income.credits
```

File: policyengine_us/variables/gov/states/oh/tax/income/credits/oh_non_public_school_credits.py [hunk 13]
```diff
@@ -13,10 +13,10 @@ class oh_non_public_school_credits(Variable):
...
 
-        agi = tax_unit("oh_agi", period)
+        agi = tax_unit("adjusted_gross_income", period)
         person = tax_unit.members
-        tuitions = sum(person("non_public_school_tuition", period))
-        eligible = tuitions > 0
-        return rates.calc(agi) * eligible
+        tuition = tax_unit.sum(person("non_public_school_tuition", period))
+        cap = rates.calc(agi)
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/oh/tax/income/credits/oh_non_public_school_credits.py:21]:
```suggestion
        cap = p.non_public_tuition.calc(agi)
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**